basic-steps/src/main/java/org/jenkinsci/plugins/workflow/steps/MailStep.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2013-2014, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.workflow.steps;
+
+import com.google.inject.Inject;
+import hudson.Extension;
+import hudson.model.TaskListener;
+import jenkins.plugins.mailer.tasks.MimeMessageBuilder;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.mail.Address;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Transport;
+import javax.mail.internet.MimeMessage;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * Simple email sender step.
+ *
+ * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ */
+public class MailStep extends AbstractStepImpl {
+
+    @DataBoundSetter
+    public String charset;
+
+    public String subject;
+
+    public String body;
+
+    @DataBoundSetter
+    public String from;
+
+    @DataBoundSetter
+    public String to;
+
+    @DataBoundSetter
+    public String cc;
+
+    @DataBoundSetter
+    public String bcc;
+
+    @DataBoundSetter
+    public String replyTo;
+
+    @DataBoundSetter
+    public String mimeType;
+
+    @DataBoundSetter
+    public String defaultSuffix;
```

Comment:
What is this field for? There is a default suffix configured in Jenkins global configuration which can be used for recipients missing `@` (a somewhat dubious feature). But if you do not want to use that suffix, you do not need this field. Just write out the full address to begin with.